lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,932 @@
+        job_id = response_output["headers"]["Location"]
+        job_id = re.search("JID_.+", job_id).group()
+        return {'ret': True, 'msg': 'Config job created', 'job_id': job_id}
+
+    def get_fan_inventory(self):
+        result = {}
+        fan_details = []
+        key = "Thermal"
+
+        # Go through list
+        for chassis_uri in self.chassis_uri_list:
+            response = self.get_request(self.root_uri + chassis_uri)
+            if response['ret'] is False:
+                return response
+            result['ret'] = True
+            data = response['data']
+            if key in data:
+                # match: found an entry for "Thermal" information = fans
+                thermal_uri = data[key]["@odata.id"]
+                response = self.get_request(self.root_uri + thermal_uri)
+                if response['ret'] is False:
+                    return response
+                result['ret'] = True
+                data = response['data']
+
+                for device in data[u'Fans']:
+                    fan = {}
+                    # There is more information available but this is most important
+                    fan['Name'] = device[u'FanName']
+                    fan['RPMs'] = device[u'Reading']
+                    fan['State'] = device[u'Status'][u'State']
+                    fan['Health'] = device[u'Status'][u'Health']
+                    fan_details.append(fan)
+                result["entries"] = fan_details
+        return result
+
+    def get_cpu_inventory(self):
+        result = {}
+        cpu_details = []
+        cpu_list = []
+        key = "Processors"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        processors_uri = data[key]["@odata.id"]
+
+        # Get a list of all CPUs and build respective URIs
+        response = self.get_request(self.root_uri + processors_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        for cpu in data[u'Members']:
+            cpu_list.append(cpu[u'@odata.id'])
+        for c in cpu_list:
+            cpu = {}
+            uri = self.root_uri + c
+            response = self.get_request(uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            cpu['Name'] = data[u'Id']
+            cpu['Manufacturer'] = data[u'Manufacturer']
+            cpu['Model'] = data[u'Model']
+            cpu['MaxSpeedMHz'] = data[u'MaxSpeedMHz']
+            cpu['TotalCores'] = data[u'TotalCores']
+            cpu['TotalThreads'] = data[u'TotalThreads']
+            cpu['State'] = data[u'Status'][u'State']
+            cpu['Health'] = data[u'Status'][u'Health']
```

Comment:
Again, building dictionary piece by piece makes little sense if we have all required data at hand.